<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Obtain the Damond_2019_Pancreas dataset — Damond_2019_Pancreas • imcdatasets</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Obtain the Damond_2019_Pancreas dataset — Damond_2019_Pancreas"><meta property="og:description" content="Obtain the Damond_2019_Pancreas dataset, which consists of three data
objects: single cell data, multichannel images and cell segmentation masks.
The data was obtained by imaging mass cytometry (IMC) of human pancreas 
sections from donors with type 1 diabetes."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">imcdatasets</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.11.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/imcdatasets.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/Guidelines_Contribution_Dataset-formatting.html">Contribution guidelines and dataset format</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/BodenmillerGroup/imcdatasets/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Obtain the Damond_2019_Pancreas dataset</h1>
    <small class="dont-index">Source: <a href="https://github.com/BodenmillerGroup/imcdatasets/blob/HEAD/R/Damond_2019_Pancreas.R" class="external-link"><code>R/Damond_2019_Pancreas.R</code></a></small>
    <div class="hidden name"><code>Damond_2019_Pancreas.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Obtain the Damond_2019_Pancreas dataset, which consists of three data
objects: single cell data, multichannel images and cell segmentation masks.
The data was obtained by imaging mass cytometry (IMC) of human pancreas 
sections from donors with type 1 diabetes.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">Damond_2019_Pancreas</span><span class="op">(</span></span>
<span>  data_type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"sce"</span>, <span class="st">"spe"</span>, <span class="st">"images"</span>, <span class="st">"masks"</span><span class="op">)</span>,</span>
<span>  full_dataset <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  version <span class="op">=</span> <span class="st">"latest"</span>,</span>
<span>  metadata <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  on_disk <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  h5FilesPath <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  force <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>data_type</dt>
<dd><p>type of object to load, `images` for multichannel images or
`masks` for cell segmentation masks. Single cell data are retrieved using 
either `sce` for the <code>SingleCellExperiment</code> format or `spe` for the  
<code>SpatialExperiment</code> format.</p></dd>


<dt>full_dataset</dt>
<dd><p>if FALSE (default), a subset corresponding to 100 images
is returned. If TRUE, the full dataset (corresponding to 845 images) is 
returned. Due to memory space limitations, this option is only available for 
single cell data and masks, not for <code>data_type = "images"</code>.</p></dd>


<dt>version</dt>
<dd><p>dataset version. By default, the latest version is returned.</p></dd>


<dt>metadata</dt>
<dd><p>if FALSE (default), the data object selected in 
<code>data_type</code> is returned. If TRUE, only the metadata associated to this
object is returned.</p></dd>


<dt>on_disk</dt>
<dd><p>logical indicating if images in form of
<a href="https://rdrr.io/pkg/HDF5Array/man/HDF5Array-class.html" class="external-link">HDF5Array</a> objects (as .h5 files) should be stored on disk
rather than in memory. This setting is valid when downloading <code>images</code>
and <code>masks</code>.</p></dd>


<dt>h5FilesPath</dt>
<dd><p>path to where the .h5 files for on disk representation
are stored. This path needs to be defined when <code>on_disk = TRUE</code>.
When files should only temporarily be stored on disk, please set
<code>h5FilesPath = getHDF5DumpDir()</code>.</p></dd>


<dt>force</dt>
<dd><p>logical indicating if images should be overwritten when files
with the same name already exist on disk.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A SingleCellExperiment object with single cell data, a
SpatialExperiment object with single cell data, a 
CytoImageList object containing multichannel images, or a
CytoImageList object containing cell segmentation masks.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>This is an Imaging Mass Cytometry (IMC) dataset from Damond et al. (2019):</p><ul><li><p><code>images</code> contains a hundred 38-channel
    images in the form of a CytoImageList class object.</p></li>
<li><p><code>masks</code> contains the cell segmentation
    masks associated with the images, in the form of a
    CytoImageList class object.</p></li>
<li><p><code>sce</code> contains the single cell data extracted from the 
    multichannel images using the cell segmentation masks, as well as the 
    associated metadata, in the form of a 
    SingleCellExperiment. This represents a total of 252,059 
    cells x 38 channels.</p></li>
<li><p><code>spe</code> same single cell data as for <code>sce</code>, but in the
    SpatialExperiment format.</p></li>
</ul><p>All data are downloaded from ExperimentHub and cached for local re-use.</p>
<p>Mapping between the three data objects is performed via variables located in
their metadata columns: <code>mcols()</code> for the CytoImageList
objects and <code>ColData()</code> for the SingleCellExperiment and 
SpatialExperiment objects. Mapping at the image level can be 
performed with the <code>image_name</code> or <code>image_number</code> variables. 
Mapping between cell segmentation masks and single cell data is performed 
with the <code>cell_number</code> variable, the values of which correspond to the 
intensity values of the <code>masks</code> object. For practical
examples, please refer to the "Accessing IMC datasets" vignette.</p>
<p>This dataset is a subset of the complete Damond et al. (2019) dataset
comprising the data from three pancreas donors at different stages of type 1
diabetes (T1D). The three donors present clearly diverging characteristics 
in terms of cell type composition and cell-cell interactions, which makes 
this dataset ideal for benchmarking spatial and neighborhood analysis 
algorithms. If <code>full_dataset = TRUE</code>, the full dataset (845 images from
12 patients) is returned. This option is not available for multichannel 
images.</p>
<p>The <code>assay</code> slots of the SingleCellExperiment and 
SpatialExperiment objects contain three assays:</p><ul><li><p><code>counts</code> contains raw mean ion counts per cell.</p></li>
<li><p><code>exprs</code> contains arsinh-transformed counts, with cofactor 1.</p></li>
<li><p><code>quant_norm</code> contains counts censored at the 99th percentile 
    and scaled 0-1.</p></li>
</ul><p>The marker-associated metadata, including antibody information and metal 
tags are stored in the <code>rowData</code> of the 
SingleCellExperiment / SpatialExperiment 
objects.</p>
<p>The cell-associated metadata are stored in the <code>colData</code> of the
SingleCellExperiment and SpatialExperiment 
objects. These metadata include cell types (in 
<code>colData(sce)$cell_type</code>) and broader cell categories, such  as
"immune" or "islet" cells (in <code>colData(sce)$cell_category</code>). In
addition, for cells located inside pancreatic islets, the islet they belong 
to is indicated in <code>colData(sce)$islet_parent</code>. For cells not located 
in islets, the "islet_parent" value is set to 0 but the spatially closest 
islet can be identified with <code>colData(sce)$islet_closest</code>.</p>
<p>The donor-associated metadata are also stored in the <code>colData</code> of the
SingleCellExperiment and SpatialExperiment 
objects. For instance, the donors' IDs can be retrieved with 
<code>colData(sce)$patient_id</code> and the donors' disease stage can be obtained
with <code>colData(sce)$patient_stage</code>.</p>
<p>Neighborhood information, defined here as cells that are localized next to 
each other, is stored as a <code>SelfHits</code> object in the <code>colPairs</code> 
slot of the <code>SingleCellExperiment</code> and SpatialExperiment 
objects.</p>
<p>The three donors in the subset present the following characteristics:</p><ul><li><p><code>6126</code> is a non-diabetic donor, with large islets containing
    many beta cells, severe infiltration of the exocrine pancreas with
    myeloid cells but limited infiltration of islets.</p></li>
<li><p><code>6414</code> is a donor with recent T1D onset (shortly after
    diagnosis) showing partial beta cell destruction and mild infiltration 
    of islets with T cells.</p></li>
<li><p><code>6180</code> is a donor with long-duration T1D (11 years after
    diagnosis), showing near-total beta cell destruction and limited immune
    cell infiltration in both the islets and the pancreas.</p></li>
</ul><p>For information about other donors in the full dataset, please refer to the
Damond et al. publication.</p>
<p>Dataset versions: a <code>version</code> argument can be passed to the function to 
specify which dataset version should be retrieved.</p><ul><li><p><code>`v0`</code>: original version (Bioconductor &lt;= 3.15).</p></li>
<li><p><code>`v1`</code>: consistent object formatting across datasets.</p></li>
</ul><p>File sizes:</p><ul><li><p><code>`images`</code>: size in memory = 7.4 Gb, size on disk = 1.7 Gb.</p></li>
<li><p><code>`masks`</code>: size in memory = 200 Mb, size on disk = 8.2 Mb.</p></li>
<li><p><code>`sce`</code>: size in memory = 353 Mb, size on disk = 204 Mb.</p></li>
<li><p><code>`spe`</code>: size in memory = 372 Mb, size on disk = 205 Mb.</p></li>
<li><p><code>`sce_full`</code>: size in memory = 2.4 Gb, size on disk = 1.5 Gb.</p></li>
<li><p><code>`spe_full`</code>: size in memory = 2.5 Gb, size on disk = 1.5 Gb.</p></li>
<li><p><code>`masks_full`</code>: size in memory = 1.4 Gb, 
    size on disk = 60 Mb.</p></li>
</ul><p>When storing images on disk, these need to be first fully read into memory
before writing them to disk. This means the process of downloading the data
is slower than directly keeping them in memory. However, downstream analysis
will lose its memory overhead when storing images on disk.</p>
<p>Original source: Damond et al. (2019):
https://doi.org/10.1016/j.cmet.2018.11.014</p>
<p>Original link to raw data, also containing the entire dataset:
https://data.mendeley.com/datasets/cydmwsfztj/2</p>
    </div>
    <div id="references">
    <h2>References</h2>
    <p>Damond N et al. (2019).
A Map of Human Type 1 Diabetes Progression by Imaging Mass Cytometry.
<em>Cell Metab</em> 29(3), 755-768.</p>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Nicolas Damond</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># Load single cell data</span></span></span>
<span class="r-in"><span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu">Damond_2019_Pancreas</span><span class="op">(</span>data_type <span class="op">=</span> <span class="st">"sce"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> snapshotDate(): 2023-10-24</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> see ?imcdatasets and browseVignettes('imcdatasets') for documentation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading from cache</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> class: SingleCellExperiment </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> dim: 38 252059 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> metadata(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> assays(3): counts exprs quant_norm</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rownames(38): H3 SMA ... DNA1 DNA2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowData names(6): channel metal ... antibody_clone full_name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colnames(252059): 138_1 138_2 ... 319_1149 319_1150</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colData names(28): cell_id image_name ... patient_ethnicity patient_BMI</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> reducedDimNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> mainExpName: Damond_2019_Pancreas_v1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> altExpNames(0):</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Display metadata</span></span></span>
<span class="r-in"><span><span class="fu">Damond_2019_Pancreas</span><span class="op">(</span>data_type <span class="op">=</span> <span class="st">"sce"</span>, metadata <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> snapshotDate(): 2023-10-24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExperimentHub with 1 record</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> # snapshotDate(): 2023-10-24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> # names(): EH7719</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> # package(): imcdatasets</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> # $dataprovider: University of Zurich</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> # $species: Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> # $rdataclass: SingleCellExperiment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> # $rdatadateadded: 2022-10-17</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> # $title: Damond_2019_Pancreas - sce - v1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> # $description: Single cell data for the Damond_2019_Pancreas IMC dataset</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> # $taxonomyid: 9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> # $genome: NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> # $sourcetype: Zip</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> # $sourceurl: http://dx.doi.org/10.17632/cydmwsfztj.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> # $sourcesize: NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> # $tags: c("SingleCellData", "TechnologyData", "Tissue") </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> # retrieve record with 'object[["EH7719"]]' </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Load masks on disk</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/HDF5Array" class="external-link">HDF5Array</a></span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>package 'HDF5Array' was built under R version 4.3.2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: DelayedArray</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: Matrix</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>package 'Matrix' was built under R version 4.3.3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: 'Matrix'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from 'package:S4Vectors':</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     expand</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: S4Arrays</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>package 'S4Arrays' was built under R version 4.3.2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: abind</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: 'abind'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from 'package:EBImage':</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     abind</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: 'S4Arrays'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from 'package:abind':</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     abind</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from 'package:EBImage':</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     abind</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from 'package:base':</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowsum</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: SparseArray</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>package 'SparseArray' was built under R version 4.3.2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: 'DelayedArray'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from 'package:base':</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     apply, scale, sweep</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: rhdf5</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>package 'rhdf5' was built under R version 4.3.2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: 'HDF5Array'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from 'package:rhdf5':</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     h5ls</span>
<span class="r-in"><span><span class="va">masks</span> <span class="op">&lt;-</span> <span class="fu">Damond_2019_Pancreas</span><span class="op">(</span>data_type <span class="op">=</span> <span class="st">"masks"</span>, on_disk <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>h5FilesPath <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/HDF5Array/man/dump-management.html" class="external-link">getHDF5DumpDir</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> snapshotDate(): 2023-10-24</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> see ?imcdatasets and browseVignettes('imcdatasets') for documentation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading from cache</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">masks</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> CytoImageList containing 6 image(s)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> names(6): E02 E03 E04 E05 E06 E07 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Each image contains 1 channel</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Nicolas Damond.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

